### PR TITLE
Changed Placement of CloseModal()

### LINKED ISSUE #2: Bug in Edit Placement.
While adding company in Edit Placement, Request to edit placement is submitted automatically, which closes the edit placement window.

### PR DESCRIPTION
**Pull request by eshwanthkartitr**
Hi @Abhinav-ark I have changed the closeModal() position and i was not able to accidently exit out of the edit placement can you verify it once bro

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "chart.js": "^4.4.1",
     "material-icons": "^1.13.12",
     "next": "^13.5.4",
+    "placement_tracker_v2": "file:",
```
Comment (Abhinav-ark) [inline package.json]: Can you comment on the necessity of this dependency and remove it if not deemed necessary?

File: package-lock.json [hunk 2]
```diff
@@ -3289,6 +3290,10 @@
         "node": ">= 6"
       }
     },
+    "node_modules/placement_tracker_v2": {
+      "resolved": "",
+      "link": true
+    },
```
Comment (Abhinav-ark) [inline package-lock.json:4577]: Can you comment on the necessity of this dependency and remove it if not deemed necessary?

**Review by Abhinav-ark — CHANGES_REQUESTED** [review]
I am getting an error in the file, `app/dashboard/admin/student/[studentId]/editPlacement/[placementID]/page.js` 

image

There might be a problem with secure local storage not getting set properly in the previous page. I'd appreciate if you look into this.

**Comment by Abhinav-ark** [comment]
Resolves #2 .

**Comment by eshwanthkartitr** [comment]
Hey @Abhinav-ark I have changed 
const isValidJobLocation = jobLocation.length > 0 to 
const isValidJobLocation = jobLocation && jobLocation.length > 0 ? true : false;
because the joblocation was null 
image

also i have added a joblocationMessage Variable
const jobLocationMessage = isValidJobLocation ? jobLocation : "";
which will replace with "" if the flasg was false
image

image


is this fine ?

File: package.json (hunk 1 above)
Comment (eshwanthkartitr) [reply package.json]: This was unnecessary i have removed it

File: package-lock.json (hunk 2 above)
Comment (eshwanthkartitr) [reply package-lock.json:4577]: This was unnecessary i have removed it

**Comment by eshwanthkartitr** [comment]
Hey @Abhinav-ark @Ashrockzzz2003  is there some other change which i need to do?

**Review by Ashrockzzz2003 — CHANGES_REQUESTED** [review]
https://github.com/user-attachments/assets/9a82afcc-bbee-4924-9e9b-ccf349ed86d4

I tried running it and the bug is still there @eshwanthkartitr 

Test it by running the backend code locally. Link to the same is available in `README.md`. Checkout #7 for test credentials. To run backend code, some env vars need to be set. Refer an issue that talks about it there as well.

Good job on customizing the catch block messages.

**Comment by eshwanthkartitr** [comment]
Hi @Ashrockzzz2003 I have made the changes now 
image
can you check now

**Review by Ashrockzzz2003 — CHANGES_REQUESTED** [review]
That bug is now fixed. But new problem. 

The newly added company must be selected in the dropdown. That's not happening. Also, after adding a new company, if I try to select a different company from the dropdown, not happening.


Build errors. `npm run build`. It was building fine before this.
image

**Comment by eshwanthkartitr** [comment]
Hi @Ashrockzzz2003 I have made the fixes and `npm run build` is running without any problem and there was a bug which was caused by usestate i have resolved it now

File: app/dashboard/admin/student/new/page.js [hunk 3]
```diff
@@ -82,19 +82,19 @@ export default function RegisterStudent() {
...
+            "studentName": studentName,
+            "studentPassword": hashPassword(studentPassword),
+            "studentSection": studentSection,
+            "studentGender": studentGender === "Male" ? "M" : studentGender === "Female" ? "F" : "O",
+            "studentBatch": studentBatch,
+            "studentDept": studentDept,
+            "isHigherStudies": isHigherStudies === "Yes" ? "1" : "0",
+            "isPlaced": isPlaced,
+            "CGPA": CGPA
+        });
```
Comment (Ashrockzzz2003) [inline app/dashboard/admin/student/new/page.js]: comment this out. needed only during dev for test.

File: app/dashboard/admin/page.js [hunk 4]
```diff
@@ -68,6 +68,7 @@ export default function AdminDashboard() {
                 "Authorization": "Bearer " + secureLocalStorage.getItem("userAccess"),
             }
         }).then((res) => {
+            console.log(res);
```
Comment (Ashrockzzz2003) [inline app/dashboard/admin/page.js]: comment this out. needed only for dev.

File: app/dashboard/admin/page.js [hunk 5]
```diff
@@ -139,10 +140,11 @@ export default function AdminDashboard() {
                     // console.log(data);
                     if (data["companyId"] !== undefined && data["companyName"] !== undefined) {
                         alertSuccess("Success", "Company added successfully.");
+                        closeModal();
                     } else {
-                        // console.log(data["companyId"]);
-                        // console.log(data["companyName"]);
-                        alertError("Error", "Something went wrong. Please try again later.");
+                        console.log(data["companyId"]);
+                        console.log(data["companyName"]);
```
Comment (Ashrockzzz2003) [inline app/dashboard/admin/page.js]: comment out

File: app/dashboard/student/editPlacement/[placementID]/page.js [hunk 6]
```diff
@@ -1,107 +1,149 @@
...
+    export default function NewPlacementScreen() {
+
+        const router = useRouter();
+
+        const { placementID } = useParams();
+        const p = secureLocalStorage.getItem("studentPlacements");
+        const placements = JSON.parse(p);
+        var placement = {};
+        
+        console.log(placements);
```
Comment (Ashrockzzz2003) [inline app/dashboard/student/editPlacement/[placementID]/page.js]: comment out

File: app/dashboard/student/editPlacement/[placementID]/page.js [hunk 7]
```diff
@@ -387,7 +428,11 @@ export default function NewPlacementScreen() {
                                 <label className="block text-md font-medium leading-6 text-black">Company</label>
                                 <div className="mt-2">
                                     <Dropdown
-                                        value={companyId} onChange={(e) => setCompanyId(e.value || '')} options={companyList} optionLabel="companyName" optionValue='id'
+                                        value={companyId} onChange={(e) =>{
+                                            console.log('Company selected:', e.value);
+                                            setCompanyId(e.value);
+                                            console.log('CompanyId updated:', companyId);
```
Comment (Ashrockzzz2003) [inline app/dashboard/student/editPlacement/[placementID]/page.js]: comment out logs.

File: app/forgotPassword/page.js [hunk 8]
```diff
@@ -42,7 +42,7 @@ export default function ForgotPassword() {
 
     const emailRegex = new RegExp(/^[a-zA-Z0-9._-]+@(cb.students.amrita.edu|cb.amrita.edu)$/);
 
-    const isValidEmail = emailRegex.test(userEmail) || userEmail === "ashrockzzz2003@gmail.com" || userEmail === "hsheadone@gmail.com";
+    const isValidEmail = emailRegex.test(userEmail) || userEmail === "ashrockzzz2003@gmail.com" || userEmail === "hsheadone@gmail.com" || userEmail === "eshwanthkartitr@gmail.com";
```
Comment (Ashrockzzz2003) [inline app/forgotPassword/page.js]: care to remove this or let it be there? your choice. 😂

File: app/login/page.js [hunk 9]
```diff
@@ -25,7 +25,7 @@ export default function Login() {
 
     const emailRegex = new RegExp(/^[a-zA-Z0-9._-]+@(cb.students.amrita.edu|cb.amrita.edu)$/);
 
-    const isValidEmail = emailRegex.test(userEmail) || userEmail === "ashrockzzz2003@gmail.com" || userEmail === "hsheadone@gmail.com" || userEmail === "abhinavramki2@gmail.com";
+    const isValidEmail = emailRegex.test(userEmail) || userEmail === "ashrockzzz2003@gmail.com" || userEmail === "hsheadone@gmail.com" || userEmail === "abhinavramki2@gmail.com" || userEmail === "eshwanthkartitr@gmail.com";
```
Comment (Ashrockzzz2003) [inline app/login/page.js]: care to remove this 😂??

File: package.json [hunk 10]
```diff
@@ -14,7 +14,8 @@
     "aos": "^2.3.4",
     "chart.js": "^4.4.1",
     "material-icons": "^1.13.12",
-    "next": "^13.5.4",
+    "next": "^15.1.3",
+    "placement_tracker_v2": "file:",
     "primereact": "^9.6.2",
```
Comment (Ashrockzzz2003) [inline package.json:18]: `placement_tracker_v2`
remove if not needed.

**Review by Ashrockzzz2003 — CHANGES_REQUESTED** [review]
All bugs seem to be fixed! Good work.

File: app/login/page.js (hunk 9 above)
Comment (eshwanthkartitr) [reply app/login/page.js]: I was trying to check admin mode also with my mail i will remove this now

File: package.json (hunk 10 above)
Comment (eshwanthkartitr) [reply package.json:18]: ok

**Comment by eshwanthkartitr** [comment]
Hi @Ashrockzzz2003 i have fixed the unnecessary display of logs and removed my mailid

File: app/login/page.js (hunk 9 above)
Comment (Ashrockzzz2003) [reply app/login/page.js]: Got it da.

**Review by Ashrockzzz2003 — APPROVED** [review]
LGTM

Good work @eshwanthkartitr

**Review by Abhinav-ark — APPROVED** [review]

**Pull request merged**